base/compiler/inferencestate.jl

```diff
@@ -1116,3 +1121,74 @@ function get_max_methods_for_module(mod::Module)
     max_methods < 0 && return nothing
     return max_methods
 end
+
+"""
+    Future{T}
```

Comment:
This is an interesting API for tasks I think. Makes getting a value out of one type stable and also convenient. Something like Future{T}(closure) that spawns a task that calls that and then getindex does a fetch.
We can even be fancy and have it not keep the task around once we've loaded it to be nice to the GC.

Reply:
That is interesting. Right now a Task is all of chunk of work, and a waitable object, and the result, but in v2 we probably could have made those aspects more distinct. I don't know in v1 what is possible. This is a fairly vanilla design of a common Future-like construct, so this is distinctly the Core.Compiler.Future and not an arbitrarily flexible object. Other places should probably use an AbstractRemoteRef or Distributed.Future instead. This one is very tightly coupled to the workq of Task-like closures and the corresponding `typeinf_local` scheduler.